### PR TITLE
feat: add page for type classes

### LINKED ISSUE #32: Document type classes
Tracking ticket to list the features and details that should be covered.

`InstanceError`s:
* `OverlappingInstances`
* `MismatchedSignatures`
* `MissingImplementation`
* `ExtraneousDefinition`
* `DuplicateTypeVariableOccurrence`
* `ComplexInstanceType`
* `OrphanInstance`
* `MissingSuperClassInstance`
* `LawlessSuperClass`
* `UnlawfulSignature`
* `IllegalOverride`
* `UnmarkedOverride`

`ResolutionError`s:
* `SealedClass`

`WeederError`s:
* `IllegalPrivateDeclaration`
* `IllegalTypeConstraintParameter`

`NameError`s:
* `IllegalSignature`

### PR DESCRIPTION
**Pull request by mlutze**
Fixes https://github.com/flix/programming-flix/issues/32

### CODE REVIEW

**Comment by mlutze** [comment]
@magnus-madsen Still WIP, but any comments so far?

**Comment by magnus-madsen** [comment]
If you do a pass, I will merge this in. (Drop the TODOs) :)

**Comment by magnus-madsen** [comment]
Let's go then I merge and we can switch to mdBook :)

**Comment by magnus-madsen** [comment]
You removed the package lock. Please restore it :)

**Comment by mlutze** [comment]
Package lock is still there, just much smaller. It did that when I ran `npm install`. Everything runs properly with the smaller lock file.

But I know very little about `npm` etc., so maybe I am missing something?

**Comment by magnus-madsen** [comment]
> Package lock is still there, just much smaller. It did that when I ran `npm install`. Everything runs properly with the smaller lock file.
> 
> But I know very little about `npm` etc., so maybe I am missing something?

TLDR: Never change the lockfile. Please restore it :)

Changing the lockfile is akin to changing the dependencies-- that's something I manage.

**Comment by mlutze** [comment]
I reverted the lockfile change, but that makes CI fail.
